### ISSUE TITLE
Having trouble behind corporate proxy

### ISSUE THREAD

**Issue by TobiasOtto** [vancluever/terraform-provider-acme#37]
Hi,

there seems to be a problem, when I run behind a corporate proxy.

situation:
- terraform 
- AWS
- I can setup my AWS environment with terraform
- environment variables regarding proxy are set: HTTP_PROXY, HTTPS_PROXY, NO_PROXY

error message:

```Error: Error applying plan:

1 error(s) occurred:

* acme_registration.reg: 1 error(s) occurred:

* acme_registration.reg: get directory at 'https://acme-staging.api.letsencrypt.org/directory': failed to get json "https://acme-staging.api.letsencrypt.org/directory": Get https://acme-staging.api.letsencrypt.org/directory: dial tcp 104.12
5.70.118:443: connectex: Ein Verbindungsversuch ist fehlgeschlagen, da die Gegenstelle nach einer bestimmten Zeitspanne nicht richtig reagiert hat, oder die hergestellte Verbindung war fehlerhaft, da der verbundene Host nicht reagiert hat.

Terraform does not automatically rollback in the face of errors.
```

some research:
- lego DOES use **ProxyFromEnvironment** (see: https://github.com/xenolf/lego/blob/master/acme/http.go):
```
var HTTPClient = http.Client{

 Transport: &http.Transport{
 Proxy: http.ProxyFromEnvironment,
 Dial: (&net.Dialer{
```

question:
- Maybe there could be an issue with passing the environment to lego?

Cheers,
Tobias.

**Comment by vancluever** [comment]
Hey @TobiasOtto, I've identified this was more than likely caused by https://github.com/xenolf/lego/issues/477.

We will be doing a update pre-ACME v2 to lego v0.5.0, which should address this problem.

Cheers!

**Comment by vancluever** [comment]
Hey @TobiasOtto, this should be available for release v0.6.0, which should be out shortly! Keep an eye out!
